{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from geopy.geocoders import Nominatim\n",
    "from opencage.geocoder import OpenCageGeocode\n",
    "import folium\n",
    "from folium.plugins import MarkerCluster"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IPS DESTINO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_ip_destino=pd.read_excel(r\"E:\\Cursos\\BC_Data_Science\\Repositorio\\ONLINE_DS_THEBRIDGE_V\\proyecto EDA\\DATASETS\\EXCEL\\filtrado_ips_destino_2vez.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Latitud</th>\n",
       "      <th>Longitud</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>44.397044</td>\n",
       "      <td>33.345246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>52.552495</td>\n",
       "      <td>29.578579</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Latitud   Longitud\n",
       "0  44.397044  33.345246\n",
       "1  52.552495  29.578579"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ip_destino.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 414 entries, 0 to 413\n",
      "Data columns (total 2 columns):\n",
      " #   Column    Non-Null Count  Dtype  \n",
      "---  ------    --------------  -----  \n",
      " 0   Latitud   414 non-null    float64\n",
      " 1   Longitud  414 non-null    float64\n",
      "dtypes: float64(2)\n",
      "memory usage: 6.6 KB\n"
     ]
    }
   ],
   "source": [
    "df_ip_destino.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error al obtener información de ubicación para (33.3452461, 44.3970441): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=44.3970441&lon=33.3452461&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85341FF0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (29.5785789, 52.5524947): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=52.5524947&lon=29.5785789&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85340880>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (29.5635563, 52.6519831): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=52.6519831&lon=29.5635563&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E813B4BE0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (8.16439635, 77.63251528382399): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=77.63251528382399&lon=8.16439635&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85341CF0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (61.1569801, 5.502897): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=5.502897&lon=61.1569801&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342170>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (47.0515728, -52.9076712): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=-52.9076712&lon=47.0515728&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342DA0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (54.16991705, 16.18381942143288): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=16.18381942143288&lon=54.16991705&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E84726830>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (56.0529456, 22.9094689): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=22.9094689&lon=56.0529456&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342AD0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (46.2257913, 14.1693085): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=14.1693085&lon=46.2257913&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85340760>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (34.0387054, -81.6845122): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=-81.6845122&lon=34.0387054&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853439A0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (26.2496296, 50.6565677): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=50.6565677&lon=26.2496296&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025EE8677F10>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (47.1361361, -0.5821618): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=-0.5821618&lon=47.1361361&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853428C0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (47.1749904, 2.9617379): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=2.9617379&lon=47.1749904&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342E60>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (26.2533249, 50.6550133): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=50.6550133&lon=26.2533249&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025EE8677F10>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (34.0544612, 131.5662264): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (34.7447219, 135.8211329): Must be a coordinate pair or Point\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\geopy\\point.py:472: UserWarning: Latitude normalization has been prohibited in the newer versions of geopy, because the normalized value happened to be on a different pole, which is probably not what was meant. If you pass coordinates as positional args, please make sure that the order is (latitude, longitude) or (y, x) in Cartesian terms.\n",
      "  return cls(*args)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error al obtener información de ubicación para (35.6007796, 45.4099334): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=45.4099334&lon=35.6007796&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85340490>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (46.4961656, 13.7111051): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=13.7111051&lon=46.4961656&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853404C0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (35.1294635, -90.4871381): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (49.1603238, -68.2016039): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=-68.2016039&lon=49.1603238&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85343550>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (38.557439, -90.109199): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (21.7411918, -82.7573743): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=-82.7573743&lon=21.7411918&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85343D60>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (40.8832473, -5.3938661): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=-5.3938661&lon=40.8832473&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85343820>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (46.930788, 4.0988996): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=4.0988996&lon=46.930788&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853402B0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (26.2533249, 50.6550133): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=50.6550133&lon=26.2533249&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025EE8677CD0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (34.98945, 136.0106203): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (36.1651981, 44.0243019): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=44.0243019&lon=36.1651981&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85341B70>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (26.2305213, 50.577057): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=50.577057&lon=26.2305213&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853439D0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (25.0566007, 121.5521199): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (45.9316719, 13.644146): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=13.644146&lon=45.9316719&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85340940>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (59.9289565, 10.5681066): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=10.5681066&lon=59.9289565&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342D40>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (58.1642399, 8.2588588): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=8.2588588&lon=58.1642399&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342B90>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (26.2520185, 50.6561649): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=50.6561649&lon=26.2520185&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E84727340>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (43.9506723, 4.7936757): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=4.7936757&lon=43.9506723&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E847DC2E0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (51.1265823, 1.3333752): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=1.3333752&lon=51.1265823&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85341C60>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (35.7049846, 139.6255499): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (41.0586294, -83.6688168): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=-83.6688168&lon=41.0586294&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342980>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (22.701883, 120.3676482): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (35.5952202, 45.4132078): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=45.4132078&lon=35.5952202&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853417E0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (26.2642444, 50.6201215): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=50.6201215&lon=26.2642444&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E810735E0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (55.6476951, 12.5054113): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=12.5054113&lon=55.6476951&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853436A0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (53.17089525, 15.40760649354428): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=15.40760649354428&lon=53.17089525&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85343B50>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (35.3367288, 134.1518997): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (35.5635044, 45.4070964): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=45.4070964&lon=35.5635044&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E847D02E0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (49.0064711, 10.036421): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=10.036421&lon=49.0064711&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E84725D20>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (26.1804918, 50.6154727): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=50.6154727&lon=26.1804918&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85340250>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (35.548772, 135.077835): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (34.4452778, -106.718185): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (33.998548, 134.244123): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (30.48211, 72.5579456): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=72.5579456&lon=30.48211&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85343EE0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (25.0375198, 121.5636796): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (25.0343086, 121.5078143): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (4.5852893, 114.2255827): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (36.1996247, 44.0090274): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=44.0090274&lon=36.1996247&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342CE0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (47.7894777, 3.5738659): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=3.5738659&lon=47.7894777&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85341DE0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (14.6443522, 120.9835997): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (35.5823009, 45.3943486): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=45.3943486&lon=35.5823009&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342C80>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (36.1728224, 43.9875408): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=43.9875408&lon=36.1728224&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853427A0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (25.18528295, 121.4422427982554): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (55.8215609, 23.9999865): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=23.9999865&lon=55.8215609&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025EE8677400>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (36.8252529, 54.4357651): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=54.4357651&lon=36.8252529&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85341660>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (49.1964755, -124.168832): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (24.8406022, 125.1652681): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (29.6566082, -97.6027921): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (48.9050006, 7.8631157): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=7.8631157&lon=48.9050006&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342E30>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (29.583545, 52.6410768): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=52.6410768&lon=29.583545&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85342440>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (59.0001472, 26.7328913): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=26.7328913&lon=59.0001472&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E84725810>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (60.214014, 11.920446): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=11.920446&lon=60.214014&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85340730>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (33.3526012, 44.4003953): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=44.4003953&lon=33.3526012&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853437F0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (33.3461322, 44.3959451): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=44.3959451&lon=33.3461322&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E85343B20>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (60.3408344, 25.0981369): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=25.0981369&lon=60.3408344&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E84727A90>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (35.7896881, 139.2582197): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (35.5961286, 45.4129725): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=45.4129725&lon=35.5961286&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853404C0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Error al obtener información de ubicación para (35.5689943, 45.3945484): HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /reverse?lat=45.3945484&lon=35.5689943&format=json&accept-language=es&addressdetails=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000025E853429E0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))\n",
      "Coordenadas: (41.0225106, 73.0609828037725), Resultado: None\n",
      "Coordenadas: (35.5605419, 45.4255895), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (45.8108503, 14.6676522), Resultado: مديرية مرخة السفلى, Gobernación de Shabwah, Yemen\n",
      "Coordenadas: (33.4644542, -79.140148), Resultado: None\n",
      "Coordenadas: (36.1606418, 44.0257392), Resultado: None\n",
      "Coordenadas: (26.1852289, 50.5451301), Resultado: вулиця Шевченка, Орестів, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "Coordenadas: (50.0876851, 19.9480783), Resultado: محافظة الأحساء, Provincia Oriental, Arabia Saudita\n",
      "Coordenadas: (35.9066751, -77.2947846), Resultado: None\n",
      "Error al obtener información de ubicación para (23.2396361, 120.1054368): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (25.1066041, 121.5337452): Must be a coordinate pair or Point\n",
      "Coordenadas: (42.7141469, -83.1128785), Resultado: None\n",
      "Coordenadas: (26.2314964, 50.5792348), Resultado: Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, 35360, Ucrania\n",
      "Coordenadas: (26.2314964, 50.5792348), Resultado: Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, 35360, Ucrania\n",
      "Coordenadas: (3.8503053, 11.472448), Resultado: Illo Sabon Gari, Bagudo, Estado de Kebbi, Nigeria\n",
      "Error al obtener información de ubicación para (25.1145207, 121.6449837): Must be a coordinate pair or Point\n",
      "Coordenadas: (-26.8878056, -51.9996202), Resultado: None\n",
      "Coordenadas: (53.0544908, 8.8161025), Resultado: None\n",
      "Error al obtener información de ubicación para (24.2007812, 120.6589929): Must be a coordinate pair or Point\n",
      "Coordenadas: (47.216728, 0.8128137), Resultado: None\n",
      "Error al obtener información de ubicación para (32.987938, 130.943139): Must be a coordinate pair or Point\n",
      "Coordenadas: (26.1228917, 50.5643377), Resultado: Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, Ucrania\n",
      "Coordenadas: (26.2488179, 50.6561047), Resultado: Лугова вулиця, Шпанів, Шпанівська сільська громада, Рівненський район, Óblast de Rivne, 35301, Ucrania\n",
      "Coordenadas: (26.2297787, 50.5784653), Resultado: Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, 35360, Ucrania\n",
      "Coordenadas: (50.0305503, 14.6653394), Resultado: Gobernación de Hadramaut, Yemen\n",
      "Coordenadas: (49.2503406, 7.015086949274588), Resultado: Jariiban جاريبان, Mudug, Galmudug غامدوغ, Somalia\n",
      "Coordenadas: (62.0689614, -6.6946673), Resultado: None\n",
      "Coordenadas: (36.1657104, 44.0145733), Resultado: None\n",
      "Coordenadas: (36.2173924, 44.0290576), Resultado: None\n",
      "Error al obtener información de ubicación para (33.7874819, 132.7503719): Must be a coordinate pair or Point\n",
      "Coordenadas: (26.2046622, 50.5547206), Resultado: Ільпінь, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "Coordenadas: (67.5560354, 24.2372411), Resultado: تعلقہ کیٹی بندر, ضلع ٹھٹہ, Sind, 73241, Pakistán\n",
      "Error al obtener información de ubicación para (32.8006653, 131.1247848): Must be a coordinate pair or Point\n",
      "Coordenadas: (-25.9654879, 32.5775595), Resultado: None\n",
      "Error al obtener información de ubicación para (10.7923649, 106.6870868): Must be a coordinate pair or Point\n",
      "Coordenadas: (33.33672, 44.4061562), Resultado: Sebastopol, Южный федеральный округ, Rusia\n",
      "Error al obtener información de ubicación para (10.7874426, 106.6459939): Must be a coordinate pair or Point\n",
      "Coordenadas: (29.5567885, 52.615475), Resultado: Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, 247410, Belarús\n",
      "Coordenadas: (26.2305213, 50.577057), Resultado: Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, 35360, Ucrania\n",
      "Coordenadas: (43.5392004, 6.6519776), Resultado: Horshagah, Shabelle, Somali, Etiopía\n",
      "Coordenadas: (36.7598616, 2.9970277), Resultado: Loiyangalani ward, Condado de Marsabit, Eastern, Kenia\n",
      "Error al obtener información de ubicación para (34.6960135, 135.9725368): Must be a coordinate pair or Point\n",
      "Coordenadas: (10.4651024, -73.24278640036832), Resultado: None\n",
      "Coordenadas: (26.1852289, 50.5451301), Resultado: вулиця Шевченка, Орестів, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "Error al obtener información de ubicación para (25.0268441, 121.550952): Must be a coordinate pair or Point\n",
      "Coordenadas: (36.1897688, 44.0049536), Resultado: None\n",
      "Coordenadas: (36.1968127, 43.9998023), Resultado: None\n",
      "Coordenadas: (35.5665568, 45.4576695), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (40.5659968, 37.6742979), Resultado: Yaprakbaşı, Çınar, Diyarbakir, Güneydoğu Anadolu Bölgesi, Turquía\n",
      "Coordenadas: (38.5273499, 35.5262433), Resultado: بئر العمالة, ناحية المنصورة, منطقة الثورة, Gobernación de Al-Raqa, Siria\n",
      "Coordenadas: (48.0056622, 5.516797), Resultado: Hobyo هبيا, Mudug, Galmudug غامدوغ, Somalia\n",
      "Coordenadas: (35.5825334, 45.4142863), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (59.3209714, 18.0680412), Resultado: None\n",
      "Coordenadas: (62.1355153, -6.8469872), Resultado: None\n",
      "Error al obtener información de ubicación para (35.0847595, 134.9570048): Must be a coordinate pair or Point\n",
      "Coordenadas: (29.5602015, 52.615277), Resultado: Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, 247410, Belarús\n",
      "Coordenadas: (49.9546785, 14.6376243), Resultado: Gobernación de Hadramaut, Yemen\n",
      "Coordenadas: (35.5702276, 45.3799918), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Error al obtener información de ubicación para (35.3758731, 134.2003824): Must be a coordinate pair or Point\n",
      "Coordenadas: (29.6839112, 52.4761999), Resultado: Асташкавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "Coordenadas: (29.5357487, 52.6240818), Resultado: Н5013, Вялікі Бор, Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, 247410, Belarús\n",
      "Coordenadas: (-24.21082035, -65.26898000858446), Resultado: None\n",
      "Coordenadas: (36.2139523, 44.0146637), Resultado: None\n",
      "Error al obtener información de ubicación para (14.5278426, 121.0047914): Must be a coordinate pair or Point\n",
      "Coordenadas: (49.197035, 4.2863591), Resultado: None\n",
      "Error al obtener información de ubicación para (34.7298607, 135.3359462): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (35.3904355, 134.1968055): Must be a coordinate pair or Point\n",
      "Coordenadas: (36.6245058, 29.1348299), Resultado: Gobernación de Tabuk, Provincia de Tabuk, Arabia Saudita\n",
      "Error al obtener información de ubicación para (25.0762981, 121.4326434): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (36.7675572, -93.3503738): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (25.0266338, 121.5479612): Must be a coordinate pair or Point\n",
      "Coordenadas: (36.1607749, 43.9814079), Resultado: None\n",
      "Coordenadas: (-37.6913531, -72.2580253), Resultado: None\n",
      "Coordenadas: (35.5843185, 45.4085846), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (35.5590011, 45.4532954), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (34.6571341, 50.8391304), Resultado: С-191541, Павлючки, Садівська сільська громада, Сумський район, Óblast de Sumy, 42345, Ucrania\n",
      "Coordenadas: (33.3524717, 60.6678335), Resultado: «Кола», Лодейнопольское городское поселение, Лодейнопольский район, Óblast de Leningrado, Северо-Западный федеральный округ, Rusia\n",
      "Coordenadas: (35.5799084, 45.4062324), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (54.8139243, 24.607217), Resultado: طريق الطَّف, مَدينة زايد العسكرية, أبوظبي, Abu Dabi, Emiratos Árabes Unidos\n",
      "Coordenadas: (37.110904, -87.886663), Resultado: None\n",
      "Error al obtener información de ubicación para (38.0223633, -103.4769085): Must be a coordinate pair or Point\n",
      "Coordenadas: (38.8422266, -79.872597), Resultado: None\n",
      "Coordenadas: (33.3525071, 44.3984031), Resultado: Sebastopol, Южный федеральный округ, Rusia\n",
      "Coordenadas: (35.598789, 45.403393), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (35.5830694, 45.3905485), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (33.3407181, 44.4062244), Resultado: Sebastopol, Южный федеральный округ, Rusia\n",
      "Error al obtener información de ubicación para (25.0296095, 121.4305389): Must be a coordinate pair or Point\n",
      "Coordenadas: (43.8245744, 5.3043759), Resultado: Afder, Somali, Etiopía\n",
      "Coordenadas: (59.9486686, 11.0423385), Resultado: None\n",
      "Coordenadas: (35.5905916, 45.4061402), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (47.2728482, -53.8395954), Resultado: None\n",
      "Coordenadas: (35.5572512, 45.4445823), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (33.3268294, 44.3599766), Resultado: Sebastopol, Южный федеральный округ, Rusia\n",
      "Coordenadas: (36.18400385, 5.445376555832292), Resultado: Kuchuru, South Omo, Naciones, Nacionalidades y Pueblos del Sur, Etiopía\n",
      "Coordenadas: (26.2488179, 50.6561047), Resultado: Лугова вулиця, Шпанів, Шпанівська сільська громада, Рівненський район, Óblast de Rivne, 35301, Ucrania\n",
      "Coordenadas: (35.5589845, 45.4577537), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (-11.8803402, -77.0165255), Resultado: None\n",
      "Coordenadas: (47.637474, -0.2750945), Resultado: None\n",
      "Coordenadas: (26.1910195, 50.5478676), Resultado: Ільпінь, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "Coordenadas: (43.042837, -74.848826), Resultado: None\n",
      "Coordenadas: (38.270887, -87.692707), Resultado: None\n",
      "Coordenadas: (35.5574604, 45.3526872), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (65.2460146, 27.5797246), Resultado: تحصیل ناگ, ضلع واشک, رخشان ڈویژن, Baluchistán, Pakistán\n",
      "Coordenadas: (35.5918495, 45.4079646), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (36.6312777, 32.920534), Resultado: عمرة, ناحية شهبا, منطقة شهبا, Gobernación de Sueida, Siria\n",
      "Coordenadas: (55.6464931, 12.16284), Resultado: None\n",
      "Coordenadas: (54.1908097, 16.2109179), Resultado: None\n",
      "Coordenadas: (46.8498882, 5.6398702), Resultado: Hogdugaag, Dhuusamarreeb دوسمريب, Galguduud, Galmudug غامدوغ, Somalia\n",
      "Error al obtener información de ubicación para (10.7931445, 106.6658884): Must be a coordinate pair or Point\n",
      "Coordenadas: (21.8798421, -82.7995103), Resultado: None\n",
      "Coordenadas: (45.2849889, 6.2974573), Resultado: Dollo, Somali, Etiopía\n",
      "Coordenadas: (49.0014272, 5.6942195), Resultado: Hobyo هبيا, Mudug, Galmudug غامدوغ, Somalia\n",
      "Coordenadas: (40.4933802, -80.6887635), Resultado: None\n",
      "Coordenadas: (35.5795906, 45.4041821), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (36.1948939, 44.0323137), Resultado: None\n",
      "Error al obtener información de ubicación para (35.9997183, 139.3875849): Must be a coordinate pair or Point\n",
      "Coordenadas: (26.1910195, 50.5478676), Resultado: Ільпінь, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "Error al obtener información de ubicación para (25.1069304, 121.5331912): Must be a coordinate pair or Point\n",
      "Coordenadas: (-21.0460626, 55.2788892), Resultado: None\n",
      "Error al obtener información de ubicación para (33.5526481, 131.7341971): Must be a coordinate pair or Point\n",
      "Coordenadas: (62.0142728, -6.8103807), Resultado: None\n",
      "Coordenadas: (12.4044174, -1.5681763), Resultado: Provincia de Ogooué-Lolo, Gabón\n",
      "Coordenadas: (6.3561546, 2.4030901), Resultado: None\n",
      "Coordenadas: (57.5952298, 9.9831319), Resultado: None\n",
      "Coordenadas: (32.4349846, 35.4837854), Resultado: None\n",
      "Error al obtener información de ubicación para (14.5453655, 120.9991114): Must be a coordinate pair or Point\n",
      "Coordenadas: (36.1039086, -85.8205806), Resultado: None\n",
      "Error al obtener información de ubicación para (14.6130399, 120.9765186): Must be a coordinate pair or Point\n",
      "Coordenadas: (35.5674138, 45.4605334), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (35.5689943, 45.3945484), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (12.3946623, -1.4569914), Resultado: Nzenzélé, Provincia de Ogooué-Lolo, Gabón\n",
      "Coordenadas: (36.2058046, 44.0070233), Resultado: None\n",
      "Coordenadas: (47.0556787, 12.9850193), Resultado: None\n",
      "Coordenadas: (35.5767669, 45.4239647), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (8.9845707, 38.8578823), Resultado: Italia\n",
      "Coordenadas: (49.1557981, -1.1973198), Resultado: None\n",
      "Error al obtener información de ubicación para (34.6602559, 135.4979372): Must be a coordinate pair or Point\n",
      "Coordenadas: (33.3236525, 44.4263399), Resultado: Sebastopol, Южный федеральный округ, Rusia\n",
      "Coordenadas: (41.604988, -84.656596), Resultado: None\n",
      "Coordenadas: (39.2732009, -79.5122998), Resultado: None\n",
      "Error al obtener información de ubicación para (37.1790346, -94.3425914): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (10.746184, 106.6799149): Must be a coordinate pair or Point\n",
      "Coordenadas: (26.1704837, 50.6243922), Resultado: Т-18-06, Городоцька сільська громада, Рівненський район, Óblast de Rivne, 35307, Ucrania\n",
      "Coordenadas: (48.770684, 10.125761), Resultado: Laasqoray لاسقرى, Sanaag, Somalia\n",
      "Coordenadas: (-36.5281553, -72.8265293), Resultado: None\n",
      "Error al obtener información de ubicación para (25.0343485, 121.5516356): Must be a coordinate pair or Point\n",
      "Coordenadas: (4.6657382, -74.0480715), Resultado: None\n",
      "Coordenadas: (48.2340784, -3.4143001), Resultado: None\n",
      "Coordenadas: (26.2281751, 50.5721692), Resultado: Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, Ucrania\n",
      "Error al obtener información de ubicación para (25.0672035, 121.5326983): Must be a coordinate pair or Point\n",
      "Coordenadas: (27.1089309, 57.0873521), Resultado: Rieksti, Balvu novads, Letgalia, LV-4561, Letonia\n",
      "Error al obtener información de ubicación para (33.9740926, 132.7719131): Must be a coordinate pair or Point\n",
      "Coordenadas: (26.2177128, 50.5660797), Resultado: М-06;Н-25, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35702, Ucrania\n",
      "Coordenadas: (59.3081888, 18.1120884), Resultado: None\n",
      "Coordenadas: (70.0811677, 25.0524862), Resultado: 69170, Sind, 69170, Pakistán\n",
      "Error al obtener información de ubicación para (3.0348073, 101.5013519): Must be a coordinate pair or Point\n",
      "Coordenadas: (43.0171932, 0.1769483), Resultado: Jamaame جمامة, Jubbada Hoose, Jubaland جوبالاند, Somalia\n",
      "Error al obtener información de ubicación para (34.9192532, 134.9370235): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (14.5373994, 121.0006589): Must be a coordinate pair or Point\n",
      "Coordenadas: (36.1671517, 44.0017565), Resultado: None\n",
      "Coordenadas: (34.39872075, 4.997583325000001), Resultado: Kapoeta Este, Ecuatoria Oriental, Sudán del Sur\n",
      "Coordenadas: (33.3500376, 44.3932106), Resultado: Sebastopol, Южный федеральный округ, Rusia\n",
      "Coordenadas: (-23.19277155, -46.81972214820819), Resultado: None\n",
      "Coordenadas: (22.9554421, -82.5567287), Resultado: None\n",
      "Coordenadas: (31.564535, 35.1179202), Resultado: None\n",
      "Coordenadas: (60.0562223, 19.9635932), Resultado: None\n",
      "Error al obtener información de ubicación para (25.0375198, 121.5636796): Must be a coordinate pair or Point\n",
      "Coordenadas: (21.8013726, -77.7502154), Resultado: None\n",
      "Error al obtener información de ubicación para (34.5373785, 133.3797832): Must be a coordinate pair or Point\n",
      "Coordenadas: (26.2484695, 50.6571517), Resultado: Лугова вулиця, Шпанів, Шпанівська сільська громада, Рівненський район, Óblast de Rivne, 35301, Ucrania\n",
      "Error al obtener información de ubicación para (38.084616, -92.953273): Must be a coordinate pair or Point\n",
      "Coordenadas: (35.5655666, 45.455258), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (55.78467105, 12.49354423118322), Resultado: None\n",
      "Coordenadas: (22.8323375, -82.0376706), Resultado: None\n",
      "Coordenadas: (29.5791516, 52.5315751), Resultado: Дарожная вуліца, Узнаж, Сасноваборскі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "Coordenadas: (35.5860589, 45.4087759), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (55.6121098, 24.6265437), Resultado: الفقع, العين, Abu Dabi, Emiratos Árabes Unidos\n",
      "Coordenadas: (60.4184915, 8.4694386), Resultado: None\n",
      "Error al obtener información de ubicación para (25.1071951, 121.7937009): Must be a coordinate pair or Point\n",
      "Coordenadas: (47.2099847, 3.1821081), Resultado: None\n",
      "Coordenadas: (36.211194, 44.0234644), Resultado: None\n",
      "Coordenadas: (49.2716179, 5.8650204), Resultado: None\n",
      "Coordenadas: (35.5800226, 45.418876), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (55.4950239, 22.2001572), Resultado: محافظة العديد, Provincia Oriental, 36621, Arabia Saudita\n",
      "Coordenadas: (41.7386357, -83.4555642), Resultado: None\n",
      "Coordenadas: (6.3540137, 2.3960256), Resultado: None\n",
      "Error al obtener información de ubicación para (3.6814363, 113.7428209): Must be a coordinate pair or Point\n",
      "Coordenadas: (35.5803509, 45.3917924), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Error al obtener información de ubicación para (22.7254988, 120.3294734): Must be a coordinate pair or Point\n",
      "Coordenadas: (55.7969657, 13.1527305), Resultado: None\n",
      "Coordenadas: (-11.879705, -77.0157476), Resultado: None\n",
      "Error al obtener información de ubicación para (32.787204, 130.6610534): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (25.1069304, 121.5331912): Must be a coordinate pair or Point\n",
      "Coordenadas: (41.0891541, -80.6463442), Resultado: None\n",
      "Coordenadas: (29.3291058, 72.97861616646486), Resultado: None\n",
      "Coordenadas: (36.1957347, 44.0368627), Resultado: None\n",
      "Coordenadas: (38.8011462, -77.5270138), Resultado: None\n",
      "Coordenadas: (48.6129704, -1.9902827), Resultado: None\n",
      "Error al obtener información de ubicación para (35.8298042, 139.6639498): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (25.0325518, 121.5515889): Must be a coordinate pair or Point\n",
      "Coordenadas: (47.4505262, -53.2725968), Resultado: None\n",
      "Coordenadas: (44.3420202, 21.900798), Resultado: محافظة وادي الدواسر, Provincia de Riad, Arabia Saudita\n",
      "Error al obtener información de ubicación para (37.5213151, 126.8063629): Must be a coordinate pair or Point\n",
      "Coordenadas: (29.5840085, 52.646024), Resultado: Н5013, Вялікі Бор, Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, 247410, Belarús\n",
      "Coordenadas: (42.5611423, -70.97970624783775), Resultado: None\n",
      "Coordenadas: (47.8743879, -3.0718248), Resultado: None\n",
      "Coordenadas: (33.3446021, 44.4021116), Resultado: Sebastopol, Южный федеральный округ, Rusia\n",
      "Error al obtener información de ubicación para (25.032755, 121.5305543): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (25.0994775, 121.7349904): Must be a coordinate pair or Point\n",
      "Coordenadas: (39.1590396, -6.6436306), Resultado: Kinondoni Municipal, Región de Dar es-Salam, Coastal Zone, Tanzania\n",
      "Coordenadas: (56.0499501, 23.9522155), Resultado: Buraimi, Omán\n",
      "Coordenadas: (35.5744257, 45.4518726), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (71.0980613, 25.3742887), Resultado: Chohtan Tehsil, Barmer District, Rajastán, India\n",
      "Coordenadas: (47.6426054, -2.6006119), Resultado: None\n",
      "Error al obtener información de ubicación para (33.1178815, 129.6689487): Must be a coordinate pair or Point\n",
      "Coordenadas: (35.5591302, 45.4374123), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (12.3419736, -1.5073256), Resultado: Nzambi I, Provincia de Ogooué-Lolo, Gabón\n",
      "Coordenadas: (36.1632831, 43.9805783), Resultado: None\n",
      "Error al obtener información de ubicación para (33.7635103, 135.3243415): Must be a coordinate pair or Point\n",
      "Coordenadas: (37.2784366, 32.898489), Resultado: ناحية السبع بيار, منطقة دوما, Gobernacion de Rif Dimashq, Siria\n",
      "Coordenadas: (47.7328901, -0.6825302), Resultado: None\n",
      "Coordenadas: (46.0161231, 6.1033079), Resultado: Dollo, Somali, Etiopía\n",
      "Coordenadas: (26.2619157, 50.6184955), Resultado: Рівненська міська рада, 12а, Соборна вулиця, Rivne, Рівненська міська громада, Рівненський район, Óblast de Rivne, 33017, Ucrania\n",
      "Coordenadas: (35.6586404, 51.3836496), Resultado: Мирная улица, Извеково, Нижнегридинский сельсовет, Большесолдатский район, Óblast de Kursk, Distrito federal Central, Rusia\n",
      "Coordenadas: (55.0851495, 26.2592462), Resultado: دهستان تنب, بخش تنب, شهرستان ابوموسی, Hormozgan, Irán\n",
      "Coordenadas: (33.151737, -80.5201471), Resultado: None\n",
      "Coordenadas: (-24.2465505, -65.2759703), Resultado: None\n",
      "Coordenadas: (42.4183899, 22.5262694), Resultado: محافظة المويه, Provincia de La Meca, Arabia Saudita\n",
      "Coordenadas: (36.1671527, 44.0313358), Resultado: None\n",
      "Coordenadas: (29.7152983, 52.4426159), Resultado: Н5009, Пажыхар, Асташкавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "Coordenadas: (36.1947938, 36.1350438), Resultado: Kamberli, Altınözü, Hatay, Región del Mediterráneo, Turquía\n",
      "Error al obtener información de ubicación para (34.5070264, 135.5553142): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (25.0652734, 121.5021266): Must be a coordinate pair or Point\n",
      "Coordenadas: (61.2670872, 70.28507214726412), Resultado: None\n",
      "Coordenadas: (42.776682, -89.2991584), Resultado: None\n",
      "Error al obtener información de ubicación para (35.0483802, 135.4144972): Must be a coordinate pair or Point\n",
      "Coordenadas: (46.4392479, 1.5857882), Resultado: None\n",
      "Coordenadas: (26.2622617, 50.6200924), Resultado: 14, вулиця Євгена Гребінки, Rivne, Рівненська міська громада, Рівненський район, Óblast de Rivne, 33017, Ucrania\n",
      "Coordenadas: (29.6275304, 52.5896819), Resultado: Н5019, Печышчы, Сасноваборскі сельскі Савет, Светлагорскі раён, Provincia de Gómel, 246284, Belarús\n",
      "Coordenadas: (35.5795062, 45.4073707), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (53.1001628, 17.90367734278327), Resultado: Dhofar, Omán\n",
      "Coordenadas: (26.1808301, 50.6103369), Resultado: Грунтова вулиця, Тинне, Rivne, Рівненська міська громада, Рівненський район, Óblast de Rivne, 33004, Ucrania\n",
      "Coordenadas: (47.6979498, 3.5951792), Resultado: None\n",
      "Coordenadas: (40.9672793, 28.6716258), Resultado: محافظة حائل, Provincia de Hail, Arabia Saudita\n",
      "Error al obtener información de ubicación para (24.9762008, 121.543542): Must be a coordinate pair or Point\n",
      "Coordenadas: (29.5572467, 52.6551318), Resultado: 67, Цэнтральная вуліца, Мольча, Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "Error al obtener información de ubicación para (25.064245, 121.5338503): Must be a coordinate pair or Point\n",
      "Coordenadas: (42.4884078, -71.15518692722445), Resultado: None\n",
      "Error al obtener información de ubicación para (37.119564, -96.181544): Must be a coordinate pair or Point\n",
      "Coordenadas: (35.5854785, 45.41613), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (41.0463473, 39.2438915), Resultado: Bedran, Serpmekaya, Karlıova, Bingöl, Doğu Anadolu Bölgesi, Turquía\n",
      "Coordenadas: (35.5617679, 45.4628714), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (36.1993741, 44.0126977), Resultado: None\n",
      "Coordenadas: (33.3155389, 44.4236059), Resultado: Sebastopol, Южный федеральный округ, Rusia\n",
      "Error al obtener información de ubicación para (10.8141669, 106.6838108): Must be a coordinate pair or Point\n",
      "Coordenadas: (36.1727232, 44.0007317), Resultado: None\n",
      "Error al obtener información de ubicación para (34.4452778, -106.7111308): Must be a coordinate pair or Point\n",
      "Coordenadas: (50.8537636, 2.3157629), Resultado: None\n",
      "Coordenadas: (48.7590733, 9.9304615), Resultado: Laasqoray لاسقرى, Sanaag, Somalia\n",
      "Coordenadas: (35.56402, 45.4050958), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (36.72092215, 52.73634155), Resultado: Моховское сельское поселение, Покровский район, Óblast de Oriol, Distrito federal Central, Rusia\n",
      "Error al obtener información de ubicación para (33.074566, 131.01391): Must be a coordinate pair or Point\n",
      "Coordenadas: (35.5867134, 45.3933433), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (35.5557952, 45.4385985), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Error al obtener información de ubicación para (37.8434667, 128.8160282): Must be a coordinate pair or Point\n",
      "Coordenadas: (26.2046622, 50.5547206), Resultado: Ільпінь, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "Error al obtener información de ubicación para (34.7798539, 135.4171733): Must be a coordinate pair or Point\n",
      "Coordenadas: (70.6139437, 26.050652), Resultado: Girab, Gadraroad Tehsil, Barmer District, Rajastán, India\n",
      "Coordenadas: (68.6569419, 17.7012183), Resultado: None\n",
      "Coordenadas: (25.3839115, 51.4152203), Resultado: О-031167, Лишнівка, Маневицька селищна громада, Камінь-Каширський район, Óblast de Volinia, 44610, Ucrania\n",
      "Coordenadas: (37.2658705, 32.8878607), Resultado: ناحية المشنف, منطقة سويداء, Gobernación de Sueida, Siria\n",
      "Coordenadas: (32.3456441, 50.8275583), Resultado: Ічнянська міська громада, Прилуцький район, Óblast de Chernihiv, Ucrania\n",
      "Coordenadas: (35.5831884, 45.3968931), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Error al obtener información de ubicación para (25.0574521, 121.4356351): Must be a coordinate pair or Point\n",
      "Coordenadas: (56.2578638, 22.4572161), Resultado: Ad Dhahirah, Omán\n",
      "Error al obtener información de ubicación para (35.3308948, 134.1075466): Must be a coordinate pair or Point\n",
      "Coordenadas: (35.5954262, 45.41022), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Error al obtener información de ubicación para (25.0684938, 121.4895826): Must be a coordinate pair or Point\n",
      "Coordenadas: (36.1752309, 43.9943456), Resultado: None\n",
      "Coordenadas: (-32.0241997, -63.0780839), Resultado: None\n",
      "Coordenadas: (26.1724017, 50.6219339), Resultado: Городоцька сільська громада, Рівненський район, Óblast de Rivne, 35307, Ucrania\n",
      "Coordenadas: (36.1640824, 44.0263868), Resultado: None\n",
      "Coordenadas: (48.4045589, 0.1956628), Resultado: None\n",
      "Coordenadas: (42.9436545, 13.6268662), Resultado: مديرية الخوخة, Gobernación de Al Hudayda, Yemen\n",
      "Error al obtener información de ubicación para (25.1120491, 121.6449183): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (25.0426085, 121.5488495): Must be a coordinate pair or Point\n",
      "Coordenadas: (35.5909164, 45.4166), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (35.5807603, 45.4282067), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (49.2339186, 7.009954715740743), Resultado: Jariiban جاريبان, Mudug, Galmudug غامدوغ, Somalia\n",
      "Coordenadas: (48.7988844, 2.0450613), Resultado: None\n",
      "Coordenadas: (35.5947006, 45.410377), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (47.5869467, -52.8677893), Resultado: None\n",
      "Coordenadas: (-29.1180167, -59.6834705), Resultado: None\n",
      "Error al obtener información de ubicación para (20.9933134, 105.829897): Must be a coordinate pair or Point\n",
      "Coordenadas: (63.0358731, 27.7048906), Resultado: تحصیل ماشکیل, ضلع واشک, رخشان ڈویژن, Baluchistán, Pakistán\n",
      "Coordenadas: (29.5639613, 52.6539897), Resultado: 32, Цэнтральная вуліца, Мольча, Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "Coordenadas: (35.7001352, 51.4823164), Resultado: 2-я Косторная, Любимовский сельсовет, Большесолдатский район, Óblast de Kursk, Distrito federal Central, Rusia\n",
      "Coordenadas: (35.5655811, 45.4706289), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (55.6710437, 12.4559913), Resultado: None\n",
      "Coordenadas: (-33.4523268, -64.760974), Resultado: None\n",
      "Error al obtener información de ubicación para (36.9557332, 126.4214965): Must be a coordinate pair or Point\n",
      "Coordenadas: (67.1588427, 26.9106891), Resultado: تحصیل خضدار, ضلع خضدار, قلات ڈویژن, Baluchistán, 89100, Pakistán\n",
      "Coordenadas: (54.1786889, 15.601285), Resultado: None\n",
      "Coordenadas: (40.9937517, -72.2349699), Resultado: None\n",
      "Error al obtener información de ubicación para (25.0274799, 121.5480283): Must be a coordinate pair or Point\n",
      "Coordenadas: (45.46594725, 9.19641986105449), Resultado: Burco برعو, Togdheer, Somalia\n",
      "Error al obtener información de ubicación para (34.4460102, -106.7128554): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (10.7740434, 106.6389914): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (25.0292927, 121.5306152): Must be a coordinate pair or Point\n",
      "Coordenadas: (60.5623837, 9.0365561), Resultado: None\n",
      "Coordenadas: (35.5770118, 45.3878112), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (26.2294212, 50.574391), Resultado: Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, Ucrania\n",
      "Coordenadas: (50.8610779, 12.8618085), Resultado: None\n",
      "Coordenadas: (26.2267376, 50.5776798), Resultado: Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, 35360, Ucrania\n",
      "Coordenadas: (46.2867072, 15.0403341), Resultado: مديرية مرخة السفلى, Gobernación de Shabwah, Yemen\n",
      "Coordenadas: (35.5643188, 45.4522615), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Coordenadas: (29.1292534, 54.0437385), Resultado: Дзмітравіцкі сельскі Савет, Бярэзінскі раён, Provincia de Minsk, Belarús\n",
      "Coordenadas: (26.1847226, 50.5453489), Resultado: вулиця Шевченка, Орестів, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "Coordenadas: (35.5825334, 45.4142863), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Error al obtener información de ubicación para (49.7020359, -104.6034749): Must be a coordinate pair or Point\n",
      "Coordenadas: (26.24825, 50.6551214), Resultado: Лугова вулиця, Шпанів, Шпанівська сільська громада, Рівненський район, Óblast de Rivne, 35301, Ucrania\n",
      "Coordenadas: (29.8153015, 52.3919521), Resultado: Хатыні, Чкалаўскі сельскі Савет, Raión de Kalinkavichy, Provincia de Gómel, Belarús\n",
      "Error al obtener información de ubicación para (25.032755, 121.5305543): Must be a coordinate pair or Point\n",
      "Coordenadas: (58.1523796, 8.0092932), Resultado: None\n",
      "Coordenadas: (26.2618072, 50.6207833), Resultado: 16, вулиця Євгена Гребінки, Rivne, Рівненська міська громада, Рівненський район, Óblast de Rivne, 33017, Ucrania\n",
      "Coordenadas: (35.5697774, 45.378956), Resultado: República Autónoma de Crimea, Ucrania\n",
      "Error al obtener información de ubicación para (34.7582409, 135.53249): Must be a coordinate pair or Point\n",
      "Coordenadas: (26.1879728, 50.5476375), Resultado: Ільпінь, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "Coordenadas: (58.9705014, 9.8650154), Resultado: None\n",
      "Error al obtener información de ubicación para (31.4747719, -103.3717363): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (34.5383133, 134.2849145): Must be a coordinate pair or Point\n",
      "Error al obtener información de ubicación para (33.9719189, 130.9444548): Must be a coordinate pair or Point\n",
      "Coordenadas: (18.2998984, -66.2525127), Resultado: None\n",
      "Coordenadas: (32.3175722, 53.9835606), Resultado: Снегиревское сельское поселение, Шумячский район, Óblast de Smolensk, Distrito federal Central, Rusia\n",
      "Coordenadas: (33.4469177, -84.1390137), Resultado: None\n",
      "Coordenadas: (49.2535907, 7.04139814184313), Resultado: Jariiban جاريبان, Mudug, Galmudug غامدوغ, Somalia\n",
      "Coordenadas: (30.8852237, -86.9527449), Resultado: None\n",
      "Coordenadas: (36.7602569, 2.9969173), Resultado: Loiyangalani ward, Condado de Marsabit, Eastern, Kenia\n",
      "Error al obtener información de ubicación para (24.7685899, 121.0871419): Must be a coordinate pair or Point\n",
      "Coordenadas: (29.5631982, 52.5297428), Resultado: Н5011, Сасноваборскі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "Coordenadas: (-37.3198543, -72.2745067), Resultado: None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "مديرية مرخة السفلى, Gobernación de Shabwah, Yemen\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "вулиця Шевченка, Орестів, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "محافظة الأحساء, Provincia Oriental, Arabia Saudita\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, 35360, Ucrania\n",
      "Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, 35360, Ucrania\n",
      "Illo Sabon Gari, Bagudo, Estado de Kebbi, Nigeria\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, Ucrania\n",
      "Лугова вулиця, Шпанів, Шпанівська сільська громада, Рівненський район, Óblast de Rivne, 35301, Ucrania\n",
      "Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, 35360, Ucrania\n",
      "Gobernación de Hadramaut, Yemen\n",
      "Jariiban جاريبان, Mudug, Galmudug غامدوغ, Somalia\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Ільпінь, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "تعلقہ کیٹی بندر, ضلع ٹھٹہ, Sind, 73241, Pakistán\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Sebastopol, Южный федеральный округ, Rusia\n",
      "None\n",
      "Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, 247410, Belarús\n",
      "Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, 35360, Ucrania\n",
      "Horshagah, Shabelle, Somali, Etiopía\n",
      "Loiyangalani ward, Condado de Marsabit, Eastern, Kenia\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "вулиця Шевченка, Орестів, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Yaprakbaşı, Çınar, Diyarbakir, Güneydoğu Anadolu Bölgesi, Turquía\n",
      "بئر العمالة, ناحية المنصورة, منطقة الثورة, Gobernación de Al-Raqa, Siria\n",
      "Hobyo هبيا, Mudug, Galmudug غامدوغ, Somalia\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, 247410, Belarús\n",
      "Gobernación de Hadramaut, Yemen\n",
      "República Autónoma de Crimea, Ucrania\n",
      "None\n",
      "Асташкавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "Н5013, Вялікі Бор, Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, 247410, Belarús\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "None\n",
      "Gobernación de Tabuk, Provincia de Tabuk, Arabia Saudita\n",
      "None\n",
      "None\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "República Autónoma de Crimea, Ucrania\n",
      "С-191541, Павлючки, Садівська сільська громада, Сумський район, Óblast de Sumy, 42345, Ucrania\n",
      "«Кола», Лодейнопольское городское поселение, Лодейнопольский район, Óblast de Leningrado, Северо-Западный федеральный округ, Rusia\n",
      "República Autónoma de Crimea, Ucrania\n",
      "طريق الطَّف, مَدينة زايد العسكرية, أبوظبي, Abu Dabi, Emiratos Árabes Unidos\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Sebastopol, Южный федеральный округ, Rusia\n",
      "República Autónoma de Crimea, Ucrania\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Sebastopol, Южный федеральный округ, Rusia\n",
      "None\n",
      "Afder, Somali, Etiopía\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Sebastopol, Южный федеральный округ, Rusia\n",
      "Kuchuru, South Omo, Naciones, Nacionalidades y Pueblos del Sur, Etiopía\n",
      "Лугова вулиця, Шпанів, Шпанівська сільська громада, Рівненський район, Óblast de Rivne, 35301, Ucrania\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Ільпінь, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "تحصیل ناگ, ضلع واشک, رخشان ڈویژن, Baluchistán, Pakistán\n",
      "República Autónoma de Crimea, Ucrania\n",
      "عمرة, ناحية شهبا, منطقة شهبا, Gobernación de Sueida, Siria\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Hogdugaag, Dhuusamarreeb دوسمريب, Galguduud, Galmudug غامدوغ, Somalia\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Dollo, Somali, Etiopía\n",
      "Hobyo هبيا, Mudug, Galmudug غامدوغ, Somalia\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Ільпінь, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Provincia de Ogooué-Lolo, Gabón\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "República Autónoma de Crimea, Ucrania\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Nzenzélé, Provincia de Ogooué-Lolo, Gabón\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Italia\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Sebastopol, Южный федеральный округ, Rusia\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "None\n",
      "Т-18-06, Городоцька сільська громада, Рівненський район, Óblast de Rivne, 35307, Ucrania\n",
      "Laasqoray لاسقرى, Sanaag, Somalia\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, Ucrania\n",
      "None\n",
      "Rieksti, Balvu novads, Letgalia, LV-4561, Letonia\n",
      "None\n",
      "М-06;Н-25, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35702, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "69170, Sind, 69170, Pakistán\n",
      "None\n",
      "Jamaame جمامة, Jubbada Hoose, Jubaland جوبالاند, Somalia\n",
      "None\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Kapoeta Este, Ecuatoria Oriental, Sudán del Sur\n",
      "Sebastopol, Южный федеральный округ, Rusia\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Лугова вулиця, Шпанів, Шпанівська сільська громада, Рівненський район, Óblast de Rivne, 35301, Ucrania\n",
      "None\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Дарожная вуліца, Узнаж, Сасноваборскі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "República Autónoma de Crimea, Ucrania\n",
      "الفقع, العين, Abu Dabi, Emiratos Árabes Unidos\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "محافظة العديد, Provincia Oriental, 36621, Arabia Saudita\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "República Autónoma de Crimea, Ucrania\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "محافظة وادي الدواسر, Provincia de Riad, Arabia Saudita\n",
      "None\n",
      "Н5013, Вялікі Бор, Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, 247410, Belarús\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Sebastopol, Южный федеральный округ, Rusia\n",
      "None\n",
      "None\n",
      "Kinondoni Municipal, Región de Dar es-Salam, Coastal Zone, Tanzania\n",
      "Buraimi, Omán\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Chohtan Tehsil, Barmer District, Rajastán, India\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Nzambi I, Provincia de Ogooué-Lolo, Gabón\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "ناحية السبع بيار, منطقة دوما, Gobernacion de Rif Dimashq, Siria\n",
      "Coordenadas en el mar u océano\n",
      "Dollo, Somali, Etiopía\n",
      "Рівненська міська рада, 12а, Соборна вулиця, Rivne, Рівненська міська громада, Рівненський район, Óblast de Rivne, 33017, Ucrania\n",
      "Мирная улица, Извеково, Нижнегридинский сельсовет, Большесолдатский район, Óblast de Kursk, Distrito federal Central, Rusia\n",
      "دهستان تنب, بخش تنب, شهرستان ابوموسی, Hormozgan, Irán\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "محافظة المويه, Provincia de La Meca, Arabia Saudita\n",
      "Coordenadas en el mar u océano\n",
      "Н5009, Пажыхар, Асташкавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "Kamberli, Altınözü, Hatay, Región del Mediterráneo, Turquía\n",
      "None\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "14, вулиця Євгена Гребінки, Rivne, Рівненська міська громада, Рівненський район, Óblast de Rivne, 33017, Ucrania\n",
      "Н5019, Печышчы, Сасноваборскі сельскі Савет, Светлагорскі раён, Provincia de Gómel, 246284, Belarús\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Dhofar, Omán\n",
      "Грунтова вулиця, Тинне, Rivne, Рівненська міська громада, Рівненський район, Óblast de Rivne, 33004, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "محافظة حائل, Provincia de Hail, Arabia Saudita\n",
      "None\n",
      "67, Цэнтральная вуліца, Мольча, Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Bedran, Serpmekaya, Karlıova, Bingöl, Doğu Anadolu Bölgesi, Turquía\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "Sebastopol, Южный федеральный округ, Rusia\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Laasqoray لاسقرى, Sanaag, Somalia\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Моховское сельское поселение, Покровский район, Óblast de Oriol, Distrito federal Central, Rusia\n",
      "None\n",
      "República Autónoma de Crimea, Ucrania\n",
      "República Autónoma de Crimea, Ucrania\n",
      "None\n",
      "Ільпінь, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "None\n",
      "Girab, Gadraroad Tehsil, Barmer District, Rajastán, India\n",
      "Coordenadas en el mar u océano\n",
      "О-031167, Лишнівка, Маневицька селищна громада, Камінь-Каширський район, Óblast de Volinia, 44610, Ucrania\n",
      "ناحية المشنف, منطقة سويداء, Gobernación de Sueida, Siria\n",
      "Ічнянська міська громада, Прилуцький район, Óblast de Chernihiv, Ucrania\n",
      "República Autónoma de Crimea, Ucrania\n",
      "None\n",
      "Ad Dhahirah, Omán\n",
      "None\n",
      "República Autónoma de Crimea, Ucrania\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "Городоцька сільська громада, Рівненський район, Óblast de Rivne, 35307, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "مديرية الخوخة, Gobernación de Al Hudayda, Yemen\n",
      "None\n",
      "None\n",
      "República Autónoma de Crimea, Ucrania\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Jariiban جاريبان, Mudug, Galmudug غامدوغ, Somalia\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "تحصیل ماشکیل, ضلع واشک, رخشان ڈویژن, Baluchistán, Pakistán\n",
      "32, Цэнтральная вуліца, Мольча, Чыркавіцкі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "2-я Косторная, Любимовский сельсовет, Большесолдатский район, Óblast de Kursk, Distrito federal Central, Rusia\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "تحصیل خضدار, ضلع خضدار, قلات ڈویژن, Baluchistán, 89100, Pakistán\n",
      "Coordenadas en el mar u océano\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "Burco برعو, Togdheer, Somalia\n",
      "None\n",
      "None\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "Великоомелянська сільська громада, Рівненський район, Óblast de Rivne, 35360, Ucrania\n",
      "مديرية مرخة السفلى, Gobernación de Shabwah, Yemen\n",
      "República Autónoma de Crimea, Ucrania\n",
      "Дзмітравіцкі сельскі Савет, Бярэзінскі раён, Provincia de Minsk, Belarús\n",
      "вулиця Шевченка, Орестів, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "República Autónoma de Crimea, Ucrania\n",
      "None\n",
      "Лугова вулиця, Шпанів, Шпанівська сільська громада, Рівненський район, Óblast de Rivne, 35301, Ucrania\n",
      "Хатыні, Чкалаўскі сельскі Савет, Raión de Kalinkavichy, Provincia de Gómel, Belarús\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "16, вулиця Євгена Гребінки, Rivne, Рівненська міська громада, Рівненський район, Óblast de Rivne, 33017, Ucrania\n",
      "República Autónoma de Crimea, Ucrania\n",
      "None\n",
      "Ільпінь, Здолбунівська міська громада, Рівненський район, Óblast de Rivne, 35708, Ucrania\n",
      "Coordenadas en el mar u océano\n",
      "None\n",
      "None\n",
      "None\n",
      "Coordenadas en el mar u océano\n",
      "Снегиревское сельское поселение, Шумячский район, Óblast de Smolensk, Distrito federal Central, Rusia\n",
      "Coordenadas en el mar u océano\n",
      "Jariiban جاريبان, Mudug, Galmudug غامدوغ, Somalia\n",
      "Coordenadas en el mar u océano\n",
      "Loiyangalani ward, Condado de Marsabit, Eastern, Kenia\n",
      "None\n",
      "Н5011, Сасноваборскі сельскі Савет, Светлагорскі раён, Provincia de Gómel, Belarús\n",
      "Coordenadas en el mar u océano\n",
      "Resultados guardados en 'resultados_geolocalizacion_destino_lugares.txt'.\n"
     ]
    }
   ],
   "source": [
    "# nombre de los lugares que tiene las coordenadas\n",
    "\n",
    "def obtener_info_lugar(longitud, latitud):\n",
    "    geolocalizador = Nominatim(user_agent=\"tu_aplicacion\")\n",
    "\n",
    "    try:\n",
    "        ubicacion = geolocalizador.reverse((latitud, longitud), language='es', exactly_one=True)\n",
    "        print(f\"Coordenadas: ({longitud}, {latitud}), Resultado: {ubicacion}\")\n",
    "        if ubicacion is not None:\n",
    "            return ubicacion.address\n",
    "        else:\n",
    "            return \"Coordenadas en el mar u océano\"\n",
    "    except Exception as e:\n",
    "        print(f\"Error al obtener información de ubicación para ({longitud}, {latitud}): {e}\")\n",
    "        return None\n",
    "\n",
    "\n",
    "# Convertir las coordenadas a punto flotante\n",
    "df_ip_destino['Longitud'] = df_ip_destino['Longitud'].astype(float)\n",
    "df_ip_destino['Latitud'] = df_ip_destino['Latitud'].astype(float)\n",
    "\n",
    "# Obtener una lista de información de lugar\n",
    "info_lugar_lista = []\n",
    "for index, row in df_ip_destino.iterrows():\n",
    "    longitud, latitud = row['Longitud'], row['Latitud']\n",
    "    info_lugar = obtener_info_lugar(longitud, latitud)\n",
    "    info_lugar_lista.append(info_lugar)\n",
    "\n",
    "# Mostrar la lista de información de lugar\n",
    "for lugar in info_lugar_lista:\n",
    "    print(lugar)\n",
    "\n",
    "# Guardar la lista en un archivo de texto\n",
    "archivo_resultados = 'resultados_geolocalizacion_destino_lugares.txt'\n",
    "with open(archivo_resultados, 'w', encoding='utf-8') as file:\n",
    "    for lugar in info_lugar_lista:\n",
    "        file.write(f\"{lugar}\\n\")\n",
    "\n",
    "print(f\"Resultados guardados en '{archivo_resultados}'.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "columna_destino= pd.read_excel(r\"E:\\Cursos\\BC_Data_Science\\Repositorio\\ONLINE_DS_THEBRIDGE_V\\proyecto EDA\\DATASETS\\EXCEL\\columna_destino.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 193 entries, 0 to 246\n",
      "Data columns (total 2 columns):\n",
      " #   Column   Non-Null Count  Dtype \n",
      "---  ------   --------------  ----- \n",
      " 0   Column1  150 non-null    object\n",
      " 1   Pais     150 non-null    object\n",
      "dtypes: object(2)\n",
      "memory usage: 4.5+ KB\n"
     ]
    }
   ],
   "source": [
    "columna_destino.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 248 entries, 0 to 247\n",
      "Data columns (total 3 columns):\n",
      " #   Column   Non-Null Count  Dtype \n",
      "---  ------   --------------  ----- \n",
      " 0   Column1  205 non-null    object\n",
      " 1   Ciudad   205 non-null    object\n",
      " 2   Pais     205 non-null    object\n",
      "dtypes: object(3)\n",
      "memory usage: 5.9+ KB\n"
     ]
    }
   ],
   "source": [
    "columna_destino.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Lista para almacenar índices de filas que cumplen con la condición\n",
    "indices_a_eliminar = []\n",
    "\n",
    "# Iterar sobre las filas del DataFrame\n",
    "for index, row in columna_destino.iterrows():\n",
    "    # Verificar la condición en la columna 'Info_Lugares'\n",
    "    if row['Column1'] == 'Coordenadas en el mar u ocÃ©ano' or row['Column1'] == \"none\":\n",
    "        # Agregar el índice a la lista\n",
    "        indices_a_eliminar.append(index)\n",
    "\n",
    "# Eliminar las filas con los índices recopilados\n",
    "columna_destino.drop(indices_a_eliminar, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                               Column1 Ciudad  \\\n",
      "0              RepÃºblica AutÃ³noma de Crimea, Ucrania          \n",
      "1    Ù…Ø¯ÙŠØ±ÙŠØ© Ù…Ø±Ø®Ø© Ø§Ù„Ø³ÙÙ„Ù‰, GobernaciÃ...          \n",
      "2    Ð²ÑƒÐ»Ð¸Ñ†Ñ Ð¨ÐµÐ²Ñ‡ÐµÐ½ÐºÐ°, ÐžÑ€ÐµÑÑ‚Ñ–Ð²,...          \n",
      "3    Ù…Ø­Ø§ÙØ¸Ø© Ø§Ù„Ø£Ø­Ø³Ø§Ø¡, Provincia Orienta...          \n",
      "5    Ð’ÐµÐ»Ð¸ÐºÐ¾Ð¾Ð¼ÐµÐ»ÑÐ½ÑÑŒÐºÐ° ÑÑ–Ð»ÑŒÑÑŒÐ...          \n",
      "..                                                 ...    ...   \n",
      "240  Ð¡Ð½ÐµÐ³Ð¸Ñ€ÐµÐ²ÑÐºÐ¾Ðµ ÑÐµÐ»ÑŒÑÐºÐ¾Ðµ Ð¿Ð¾...          \n",
      "242  Jariiban Ø¬Ø§Ø±ÙŠØ¨Ø§Ù†, Mudug, Galmudug ØºØ§Ù...          \n",
      "244  Loiyangalani ward, Condado de Marsabit, Easter...          \n",
      "245                                                NaN   None   \n",
      "246  Ð5011, Ð¡Ð°ÑÐ½Ð¾Ð²Ð°Ð±Ð¾Ñ€ÑÐºÑ– ÑÐµÐ»ÑŒÑÐ...          \n",
      "\n",
      "                                                  Pais  \n",
      "0              RepÃºblica AutÃ³noma de Crimea, Ucrania  \n",
      "1                                       Shabwah, Yemen  \n",
      "2                                              Ucrania  \n",
      "3                   Provincia Oriental, Arabia Saudita  \n",
      "5                                              Ucrania  \n",
      "..                                                 ...  \n",
      "240  Ð¿Ð¾ÑÐµÐ»ÐµÐ½Ð¸Ðµ, Smolensk, Distrito federal...  \n",
      "242                                     Mudug, Somalia  \n",
      "244                Condado de Marsabit, Eastern, Kenia  \n",
      "245                                               None  \n",
      "246                      Provincia de GÃ³mel, BelarÃºs  \n",
      "\n",
      "[193 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import spacy\n",
    "\n",
    "# Cargar el modelo de lenguaje de spaCy\n",
    "nlp = spacy.load('es_core_news_sm')\n",
    "\n",
    "\n",
    "# Función para clasificar entidades\n",
    "def clasificar_entidades(row):\n",
    "    if pd.notna(row['Column1']):  # Verificar si el valor no es nulo\n",
    "        ciudades = []\n",
    "        paises = []\n",
    "        \n",
    "        for entidad in nlp(row['Column1']).ents:\n",
    "            if entidad.label_ == 'GPE':\n",
    "                ciudades.append(entidad.text)\n",
    "            elif entidad.label_ == 'LOC':\n",
    "                paises.append(entidad.text)\n",
    "        \n",
    "        return pd.Series({'Ciudad': ', '.join(ciudades), 'Pais': ', '.join(paises)})\n",
    "    else:\n",
    "        return pd.Series({'Ciudad': None, 'Pais': None})\n",
    "\n",
    "# Aplicar la función a cada fila del DataFrame\n",
    "resultados = columna_destino.apply(clasificar_entidades, axis=1)\n",
    "\n",
    "# Combinar los resultados con el DataFrame original\n",
    "columna_destino = pd.concat([columna_destino, resultados], axis=1)\n",
    "\n",
    "# Mostrar el DataFrame resultante\n",
    "print(columna_destino[['Column1', 'Ciudad', 'Pais']])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Column1</th>\n",
       "      <th>Ciudad</th>\n",
       "      <th>Pais</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RepÃºblica AutÃ³noma de Crimea, Ucrania</td>\n",
       "      <td></td>\n",
       "      <td>RepÃºblica AutÃ³noma de Crimea, Ucrania</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ù…Ø¯ÙŠØ±ÙŠØ© Ù…Ø±Ø®Ø© Ø§Ù„Ø³ÙÙ„Ù‰, GobernaciÃ...</td>\n",
       "      <td></td>\n",
       "      <td>Shabwah, Yemen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ð²ÑƒÐ»Ð¸Ñ†Ñ Ð¨ÐµÐ²Ñ‡ÐµÐ½ÐºÐ°, ÐžÑ€ÐµÑÑ‚Ñ–Ð²,...</td>\n",
       "      <td></td>\n",
       "      <td>Ucrania</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ù…Ø­Ø§ÙØ¸Ø© Ø§Ù„Ø£Ø­Ø³Ø§Ø¡, Provincia Orienta...</td>\n",
       "      <td></td>\n",
       "      <td>Provincia Oriental, Arabia Saudita</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Ð’ÐµÐ»Ð¸ÐºÐ¾Ð¾Ð¼ÐµÐ»ÑÐ½ÑÑŒÐºÐ° ÑÑ–Ð»ÑŒÑÑŒÐ...</td>\n",
       "      <td></td>\n",
       "      <td>Ucrania</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>Ð¡Ð½ÐµÐ³Ð¸Ñ€ÐµÐ²ÑÐºÐ¾Ðµ ÑÐµÐ»ÑŒÑÐºÐ¾Ðµ Ð¿Ð¾...</td>\n",
       "      <td></td>\n",
       "      <td>Ð¿Ð¾ÑÐµÐ»ÐµÐ½Ð¸Ðµ, Smolensk, Distrito federal...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>Jariiban Ø¬Ø§Ø±ÙŠØ¨Ø§Ù†, Mudug, Galmudug ØºØ§Ù...</td>\n",
       "      <td></td>\n",
       "      <td>Mudug, Somalia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>Loiyangalani ward, Condado de Marsabit, Easter...</td>\n",
       "      <td></td>\n",
       "      <td>Condado de Marsabit, Eastern, Kenia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>Ð5011, Ð¡Ð°ÑÐ½Ð¾Ð²Ð°Ð±Ð¾Ñ€ÑÐºÑ– ÑÐµÐ»ÑŒÑÐ...</td>\n",
       "      <td></td>\n",
       "      <td>Provincia de GÃ³mel, BelarÃºs</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               Column1 Ciudad  \\\n",
       "0              RepÃºblica AutÃ³noma de Crimea, Ucrania          \n",
       "1    Ù…Ø¯ÙŠØ±ÙŠØ© Ù…Ø±Ø®Ø© Ø§Ù„Ø³ÙÙ„Ù‰, GobernaciÃ...          \n",
       "2    Ð²ÑƒÐ»Ð¸Ñ†Ñ Ð¨ÐµÐ²Ñ‡ÐµÐ½ÐºÐ°, ÐžÑ€ÐµÑÑ‚Ñ–Ð²,...          \n",
       "3    Ù…Ø­Ø§ÙØ¸Ø© Ø§Ù„Ø£Ø­Ø³Ø§Ø¡, Provincia Orienta...          \n",
       "5    Ð’ÐµÐ»Ð¸ÐºÐ¾Ð¾Ð¼ÐµÐ»ÑÐ½ÑÑŒÐºÐ° ÑÑ–Ð»ÑŒÑÑŒÐ...          \n",
       "..                                                 ...    ...   \n",
       "240  Ð¡Ð½ÐµÐ³Ð¸Ñ€ÐµÐ²ÑÐºÐ¾Ðµ ÑÐµÐ»ÑŒÑÐºÐ¾Ðµ Ð¿Ð¾...          \n",
       "242  Jariiban Ø¬Ø§Ø±ÙŠØ¨Ø§Ù†, Mudug, Galmudug ØºØ§Ù...          \n",
       "244  Loiyangalani ward, Condado de Marsabit, Easter...          \n",
       "245                                                NaN   None   \n",
       "246  Ð5011, Ð¡Ð°ÑÐ½Ð¾Ð²Ð°Ð±Ð¾Ñ€ÑÐºÑ– ÑÐµÐ»ÑŒÑÐ...          \n",
       "\n",
       "                                                  Pais  \n",
       "0              RepÃºblica AutÃ³noma de Crimea, Ucrania  \n",
       "1                                       Shabwah, Yemen  \n",
       "2                                              Ucrania  \n",
       "3                   Provincia Oriental, Arabia Saudita  \n",
       "5                                              Ucrania  \n",
       "..                                                 ...  \n",
       "240  Ð¿Ð¾ÑÐµÐ»ÐµÐ½Ð¸Ðµ, Smolensk, Distrito federal...  \n",
       "242                                     Mudug, Somalia  \n",
       "244                Condado de Marsabit, Eastern, Kenia  \n",
       "245                                               None  \n",
       "246                      Provincia de GÃ³mel, BelarÃºs  \n",
       "\n",
       "[193 rows x 3 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columna_destino"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 193 entries, 0 to 246\n",
      "Data columns (total 3 columns):\n",
      " #   Column   Non-Null Count  Dtype \n",
      "---  ------   --------------  ----- \n",
      " 0   Column1  150 non-null    object\n",
      " 1   Ciudad   150 non-null    object\n",
      " 2   Pais     150 non-null    object\n",
      "dtypes: object(3)\n",
      "memory usage: 6.0+ KB\n"
     ]
    }
   ],
   "source": [
    "columna_destino.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "columna_destino.drop(columns=(\"Ciudad\"), inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Column1</th>\n",
       "      <th>Pais</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RepÃºblica AutÃ³noma de Crimea, Ucrania</td>\n",
       "      <td>RepÃºblica AutÃ³noma de Crimea, Ucrania</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ù…Ø¯ÙŠØ±ÙŠØ© Ù…Ø±Ø®Ø© Ø§Ù„Ø³ÙÙ„Ù‰, GobernaciÃ...</td>\n",
       "      <td>Shabwah, Yemen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ð²ÑƒÐ»Ð¸Ñ†Ñ Ð¨ÐµÐ²Ñ‡ÐµÐ½ÐºÐ°, ÐžÑ€ÐµÑÑ‚Ñ–Ð²,...</td>\n",
       "      <td>Ucrania</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ù…Ø­Ø§ÙØ¸Ø© Ø§Ù„Ø£Ø­Ø³Ø§Ø¡, Provincia Orienta...</td>\n",
       "      <td>Provincia Oriental, Arabia Saudita</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Ð’ÐµÐ»Ð¸ÐºÐ¾Ð¾Ð¼ÐµÐ»ÑÐ½ÑÑŒÐºÐ° ÑÑ–Ð»ÑŒÑÑŒÐ...</td>\n",
       "      <td>Ucrania</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>Ð¡Ð½ÐµÐ³Ð¸Ñ€ÐµÐ²ÑÐºÐ¾Ðµ ÑÐµÐ»ÑŒÑÐºÐ¾Ðµ Ð¿Ð¾...</td>\n",
       "      <td>Ð¿Ð¾ÑÐµÐ»ÐµÐ½Ð¸Ðµ, Smolensk, Distrito federal...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>Jariiban Ø¬Ø§Ø±ÙŠØ¨Ø§Ù†, Mudug, Galmudug ØºØ§Ù...</td>\n",
       "      <td>Mudug, Somalia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>Loiyangalani ward, Condado de Marsabit, Easter...</td>\n",
       "      <td>Condado de Marsabit, Eastern, Kenia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>Ð5011, Ð¡Ð°ÑÐ½Ð¾Ð²Ð°Ð±Ð¾Ñ€ÑÐºÑ– ÑÐµÐ»ÑŒÑÐ...</td>\n",
       "      <td>Provincia de GÃ³mel, BelarÃºs</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               Column1  \\\n",
       "0              RepÃºblica AutÃ³noma de Crimea, Ucrania   \n",
       "1    Ù…Ø¯ÙŠØ±ÙŠØ© Ù…Ø±Ø®Ø© Ø§Ù„Ø³ÙÙ„Ù‰, GobernaciÃ...   \n",
       "2    Ð²ÑƒÐ»Ð¸Ñ†Ñ Ð¨ÐµÐ²Ñ‡ÐµÐ½ÐºÐ°, ÐžÑ€ÐµÑÑ‚Ñ–Ð²,...   \n",
       "3    Ù…Ø­Ø§ÙØ¸Ø© Ø§Ù„Ø£Ø­Ø³Ø§Ø¡, Provincia Orienta...   \n",
       "5    Ð’ÐµÐ»Ð¸ÐºÐ¾Ð¾Ð¼ÐµÐ»ÑÐ½ÑÑŒÐºÐ° ÑÑ–Ð»ÑŒÑÑŒÐ...   \n",
       "..                                                 ...   \n",
       "240  Ð¡Ð½ÐµÐ³Ð¸Ñ€ÐµÐ²ÑÐºÐ¾Ðµ ÑÐµÐ»ÑŒÑÐºÐ¾Ðµ Ð¿Ð¾...   \n",
       "242  Jariiban Ø¬Ø§Ø±ÙŠØ¨Ø§Ù†, Mudug, Galmudug ØºØ§Ù...   \n",
       "244  Loiyangalani ward, Condado de Marsabit, Easter...   \n",
       "245                                                NaN   \n",
       "246  Ð5011, Ð¡Ð°ÑÐ½Ð¾Ð²Ð°Ð±Ð¾Ñ€ÑÐºÑ– ÑÐµÐ»ÑŒÑÐ...   \n",
       "\n",
       "                                                  Pais  \n",
       "0              RepÃºblica AutÃ³noma de Crimea, Ucrania  \n",
       "1                                       Shabwah, Yemen  \n",
       "2                                              Ucrania  \n",
       "3                   Provincia Oriental, Arabia Saudita  \n",
       "5                                              Ucrania  \n",
       "..                                                 ...  \n",
       "240  Ð¿Ð¾ÑÐµÐ»ÐµÐ½Ð¸Ðµ, Smolensk, Distrito federal...  \n",
       "242                                     Mudug, Somalia  \n",
       "244                Condado de Marsabit, Eastern, Kenia  \n",
       "245                                               None  \n",
       "246                      Provincia de GÃ³mel, BelarÃºs  \n",
       "\n",
       "[193 rows x 2 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columna_destino"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "ename": "GeocoderUnavailable",
     "evalue": "HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rep%C3%83%C2%BAblica+Aut%C3%83%C2%B3noma+de+Crimea%2C+Ucrania&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000026FFE4978B0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTimeoutError\u001b[0m                              Traceback (most recent call last)",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\connection.py:174\u001b[0m, in \u001b[0;36mHTTPConnection._new_conn\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    173\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 174\u001b[0m     conn \u001b[38;5;241m=\u001b[39m connection\u001b[38;5;241m.\u001b[39mcreate_connection(\n\u001b[0;32m    175\u001b[0m         (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_dns_host, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mport), \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtimeout, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mextra_kw\n\u001b[0;32m    176\u001b[0m     )\n\u001b[0;32m    178\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m SocketTimeout:\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\util\\connection.py:95\u001b[0m, in \u001b[0;36mcreate_connection\u001b[1;34m(address, timeout, source_address, socket_options)\u001b[0m\n\u001b[0;32m     94\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m err \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m---> 95\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m err\n\u001b[0;32m     97\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m socket\u001b[38;5;241m.\u001b[39merror(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgetaddrinfo returns an empty list\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\util\\connection.py:85\u001b[0m, in \u001b[0;36mcreate_connection\u001b[1;34m(address, timeout, source_address, socket_options)\u001b[0m\n\u001b[0;32m     84\u001b[0m     sock\u001b[38;5;241m.\u001b[39mbind(source_address)\n\u001b[1;32m---> 85\u001b[0m \u001b[43msock\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[43msa\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     86\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m sock\n",
      "\u001b[1;31mTimeoutError\u001b[0m: timed out",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mConnectTimeoutError\u001b[0m                       Traceback (most recent call last)",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\connectionpool.py:715\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[0;32m    714\u001b[0m \u001b[38;5;66;03m# Make the request on the httplib connection object.\u001b[39;00m\n\u001b[1;32m--> 715\u001b[0m httplib_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    716\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    717\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    718\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    719\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout_obj\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    720\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    721\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    722\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    723\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    725\u001b[0m \u001b[38;5;66;03m# If we're going to release the connection in ``finally:``, then\u001b[39;00m\n\u001b[0;32m    726\u001b[0m \u001b[38;5;66;03m# the response doesn't need to know about the connection. Otherwise\u001b[39;00m\n\u001b[0;32m    727\u001b[0m \u001b[38;5;66;03m# it will also try to release it and we'll have a double-release\u001b[39;00m\n\u001b[0;32m    728\u001b[0m \u001b[38;5;66;03m# mess.\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\connectionpool.py:404\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[0;32m    403\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 404\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_validate_conn\u001b[49m\u001b[43m(\u001b[49m\u001b[43mconn\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    405\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (SocketTimeout, BaseSSLError) \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    406\u001b[0m     \u001b[38;5;66;03m# Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\connectionpool.py:1058\u001b[0m, in \u001b[0;36mHTTPSConnectionPool._validate_conn\u001b[1;34m(self, conn)\u001b[0m\n\u001b[0;32m   1057\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(conn, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msock\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m):  \u001b[38;5;66;03m# AppEngine might not have  `.sock`\u001b[39;00m\n\u001b[1;32m-> 1058\u001b[0m     \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1060\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_verified:\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\connection.py:363\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    361\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mconnect\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[0;32m    362\u001b[0m     \u001b[38;5;66;03m# Add certificate verification\u001b[39;00m\n\u001b[1;32m--> 363\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock \u001b[38;5;241m=\u001b[39m conn \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_new_conn\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    364\u001b[0m     hostname \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhost\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\connection.py:179\u001b[0m, in \u001b[0;36mHTTPConnection._new_conn\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    178\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m SocketTimeout:\n\u001b[1;32m--> 179\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m ConnectTimeoutError(\n\u001b[0;32m    180\u001b[0m         \u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m    181\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConnection to \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m timed out. (connect timeout=\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m)\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m    182\u001b[0m         \u001b[38;5;241m%\u001b[39m (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhost, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtimeout),\n\u001b[0;32m    183\u001b[0m     )\n\u001b[0;32m    185\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m SocketError \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "\u001b[1;31mConnectTimeoutError\u001b[0m: (<urllib3.connection.HTTPSConnection object at 0x0000026FFE4978B0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)')",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\requests\\adapters.py:486\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[1;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[0;32m    485\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 486\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murlopen\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    487\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    488\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    489\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    490\u001b[0m \u001b[43m        \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    491\u001b[0m \u001b[43m        \u001b[49m\u001b[43mredirect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m    492\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_same_host\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m    493\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m    494\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m    495\u001b[0m \u001b[43m        \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    496\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    497\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    498\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    500\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\connectionpool.py:827\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[0;32m    824\u001b[0m     log\u001b[38;5;241m.\u001b[39mwarning(\n\u001b[0;32m    825\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRetrying (\u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m) after connection broken by \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m: \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, retries, err, url\n\u001b[0;32m    826\u001b[0m     )\n\u001b[1;32m--> 827\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39murlopen(\n\u001b[0;32m    828\u001b[0m         method,\n\u001b[0;32m    829\u001b[0m         url,\n\u001b[0;32m    830\u001b[0m         body,\n\u001b[0;32m    831\u001b[0m         headers,\n\u001b[0;32m    832\u001b[0m         retries,\n\u001b[0;32m    833\u001b[0m         redirect,\n\u001b[0;32m    834\u001b[0m         assert_same_host,\n\u001b[0;32m    835\u001b[0m         timeout\u001b[38;5;241m=\u001b[39mtimeout,\n\u001b[0;32m    836\u001b[0m         pool_timeout\u001b[38;5;241m=\u001b[39mpool_timeout,\n\u001b[0;32m    837\u001b[0m         release_conn\u001b[38;5;241m=\u001b[39mrelease_conn,\n\u001b[0;32m    838\u001b[0m         chunked\u001b[38;5;241m=\u001b[39mchunked,\n\u001b[0;32m    839\u001b[0m         body_pos\u001b[38;5;241m=\u001b[39mbody_pos,\n\u001b[0;32m    840\u001b[0m         \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mresponse_kw\n\u001b[0;32m    841\u001b[0m     )\n\u001b[0;32m    843\u001b[0m \u001b[38;5;66;03m# Handle redirect?\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\connectionpool.py:827\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[0;32m    824\u001b[0m     log\u001b[38;5;241m.\u001b[39mwarning(\n\u001b[0;32m    825\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRetrying (\u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m) after connection broken by \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m: \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, retries, err, url\n\u001b[0;32m    826\u001b[0m     )\n\u001b[1;32m--> 827\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39murlopen(\n\u001b[0;32m    828\u001b[0m         method,\n\u001b[0;32m    829\u001b[0m         url,\n\u001b[0;32m    830\u001b[0m         body,\n\u001b[0;32m    831\u001b[0m         headers,\n\u001b[0;32m    832\u001b[0m         retries,\n\u001b[0;32m    833\u001b[0m         redirect,\n\u001b[0;32m    834\u001b[0m         assert_same_host,\n\u001b[0;32m    835\u001b[0m         timeout\u001b[38;5;241m=\u001b[39mtimeout,\n\u001b[0;32m    836\u001b[0m         pool_timeout\u001b[38;5;241m=\u001b[39mpool_timeout,\n\u001b[0;32m    837\u001b[0m         release_conn\u001b[38;5;241m=\u001b[39mrelease_conn,\n\u001b[0;32m    838\u001b[0m         chunked\u001b[38;5;241m=\u001b[39mchunked,\n\u001b[0;32m    839\u001b[0m         body_pos\u001b[38;5;241m=\u001b[39mbody_pos,\n\u001b[0;32m    840\u001b[0m         \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mresponse_kw\n\u001b[0;32m    841\u001b[0m     )\n\u001b[0;32m    843\u001b[0m \u001b[38;5;66;03m# Handle redirect?\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\connectionpool.py:799\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[0;32m    797\u001b[0m     e \u001b[38;5;241m=\u001b[39m ProtocolError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConnection aborted.\u001b[39m\u001b[38;5;124m\"\u001b[39m, e)\n\u001b[1;32m--> 799\u001b[0m retries \u001b[38;5;241m=\u001b[39m \u001b[43mretries\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mincrement\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    800\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43merror\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43me\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_pool\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_stacktrace\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msys\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mexc_info\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m2\u001b[39;49m\u001b[43m]\u001b[49m\n\u001b[0;32m    801\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    802\u001b[0m retries\u001b[38;5;241m.\u001b[39msleep()\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\urllib3\\util\\retry.py:592\u001b[0m, in \u001b[0;36mRetry.increment\u001b[1;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[0;32m    591\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m new_retry\u001b[38;5;241m.\u001b[39mis_exhausted():\n\u001b[1;32m--> 592\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m MaxRetryError(_pool, url, error \u001b[38;5;129;01mor\u001b[39;00m ResponseError(cause))\n\u001b[0;32m    594\u001b[0m log\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIncremented Retry for (url=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, url, new_retry)\n",
      "\u001b[1;31mMaxRetryError\u001b[0m: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rep%C3%83%C2%BAblica+Aut%C3%83%C2%B3noma+de+Crimea%2C+Ucrania&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000026FFE4978B0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mConnectTimeout\u001b[0m                            Traceback (most recent call last)",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\geopy\\adapters.py:482\u001b[0m, in \u001b[0;36mRequestsAdapter._request\u001b[1;34m(self, url, timeout, headers)\u001b[0m\n\u001b[0;32m    481\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 482\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msession\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    483\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m error:\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\requests\\sessions.py:602\u001b[0m, in \u001b[0;36mSession.get\u001b[1;34m(self, url, **kwargs)\u001b[0m\n\u001b[0;32m    601\u001b[0m kwargs\u001b[38;5;241m.\u001b[39msetdefault(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m--> 602\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mrequest(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGET\u001b[39m\u001b[38;5;124m\"\u001b[39m, url, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\requests\\sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[1;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[0;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[1;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msend(prep, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39msend_kwargs)\n\u001b[0;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\requests\\sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[1;34m(self, request, **kwargs)\u001b[0m\n\u001b[0;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[1;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m adapter\u001b[38;5;241m.\u001b[39msend(request, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\requests\\adapters.py:507\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[1;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[0;32m    506\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(e\u001b[38;5;241m.\u001b[39mreason, NewConnectionError):\n\u001b[1;32m--> 507\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m ConnectTimeout(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[0;32m    509\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(e\u001b[38;5;241m.\u001b[39mreason, ResponseError):\n",
      "\u001b[1;31mConnectTimeout\u001b[0m: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rep%C3%83%C2%BAblica+Aut%C3%83%C2%B3noma+de+Crimea%2C+Ucrania&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000026FFE4978B0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mGeocoderUnavailable\u001b[0m                       Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[14], line 21\u001b[0m\n\u001b[0;32m     18\u001b[0m \u001b[38;5;66;03m# Agregar marcadores al grupo con iconos personalizados\u001b[39;00m\n\u001b[0;32m     19\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m pais \u001b[38;5;129;01min\u001b[39;00m lista_origen:\n\u001b[0;32m     20\u001b[0m     \u001b[38;5;66;03m# Obtener las coordenadas mediante geocodificación\u001b[39;00m\n\u001b[1;32m---> 21\u001b[0m     ubicacion \u001b[38;5;241m=\u001b[39m \u001b[43mgeolocalizador\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mgeocode\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpais\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     22\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m     23\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m ubicacion:\n\u001b[0;32m     24\u001b[0m             \u001b[38;5;66;03m# Definir la ruta del icono personalizado\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\geopy\\geocoders\\nominatim.py:297\u001b[0m, in \u001b[0;36mNominatim.geocode\u001b[1;34m(self, query, exactly_one, timeout, limit, addressdetails, language, geometry, extratags, country_codes, viewbox, bounded, featuretype, namedetails)\u001b[0m\n\u001b[0;32m    295\u001b[0m logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m.geocode: \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__class__\u001b[39m\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__name__\u001b[39m, url)\n\u001b[0;32m    296\u001b[0m callback \u001b[38;5;241m=\u001b[39m partial(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_parse_json, exactly_one\u001b[38;5;241m=\u001b[39mexactly_one)\n\u001b[1;32m--> 297\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_call_geocoder\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcallback\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\geopy\\geocoders\\base.py:368\u001b[0m, in \u001b[0;36mGeocoder._call_geocoder\u001b[1;34m(self, url, callback, timeout, is_json, headers)\u001b[0m\n\u001b[0;32m    366\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    367\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m is_json:\n\u001b[1;32m--> 368\u001b[0m         result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43madapter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_json\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mreq_headers\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    369\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    370\u001b[0m         result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39madapter\u001b[38;5;241m.\u001b[39mget_text(url, timeout\u001b[38;5;241m=\u001b[39mtimeout, headers\u001b[38;5;241m=\u001b[39mreq_headers)\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\geopy\\adapters.py:472\u001b[0m, in \u001b[0;36mRequestsAdapter.get_json\u001b[1;34m(self, url, timeout, headers)\u001b[0m\n\u001b[0;32m    471\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget_json\u001b[39m(\u001b[38;5;28mself\u001b[39m, url, \u001b[38;5;241m*\u001b[39m, timeout, headers):\n\u001b[1;32m--> 472\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    473\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    474\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m resp\u001b[38;5;241m.\u001b[39mjson()\n",
      "File \u001b[1;32mc:\\Users\\victo\\AppData\\Local\\Programs\\Python\\Python310\\lib\\site-packages\\geopy\\adapters.py:494\u001b[0m, in \u001b[0;36mRequestsAdapter._request\u001b[1;34m(self, url, timeout, headers)\u001b[0m\n\u001b[0;32m    492\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m GeocoderServiceError(message)\n\u001b[0;32m    493\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m--> 494\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m GeocoderUnavailable(message)\n\u001b[0;32m    495\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(error, requests\u001b[38;5;241m.\u001b[39mTimeout):\n\u001b[0;32m    496\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m GeocoderTimedOut(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mService timed out\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[1;31mGeocoderUnavailable\u001b[0m: HTTPSConnectionPool(host='nominatim.openstreetmap.org', port=443): Max retries exceeded with url: /search?q=Rep%C3%83%C2%BAblica+Aut%C3%83%C2%B3noma+de+Crimea%2C+Ucrania&format=json&limit=1 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000026FFE4978B0>, 'Connection to nominatim.openstreetmap.org timed out. (connect timeout=1)'))"
     ]
    }
   ],
   "source": [
    "#GEOLOCALIZAMOSPOR PAIS, HABIENDO QUITADO ZONAS DE MAR Y ZONAS NO LOCALIZABLES DE DESTINO\n",
    "\n",
    "# Filtrar el DataFrame para quedarse solo con las filas que tienen información de geolocalización\n",
    "df_lugares_filtrado = columna_destino[['Pais']].dropna(subset=['Pais'])\n",
    "\n",
    "# Convertir la columna 'Pais' a una lista de ubicaciones\n",
    "lista_origen = df_lugares_filtrado['Pais'].tolist()\n",
    "\n",
    "# Crear un objeto de mapa centrado en una ubicación predeterminada\n",
    "mapa = folium.Map(location=[0, 0], zoom_start=2)\n",
    "\n",
    "# Agregar un grupo de marcadores\n",
    "marker_cluster = MarkerCluster().add_to(mapa)\n",
    "\n",
    "# Crear un objeto geocoder de Nominatim\n",
    "geolocalizador = Nominatim(user_agent=\"mi_aplicacion\")\n",
    "\n",
    "# Agregar marcadores al grupo con iconos personalizados\n",
    "for pais in lista_origen:\n",
    "    # Obtener las coordenadas mediante geocodificación\n",
    "    ubicacion = geolocalizador.geocode(pais)\n",
    "    try:\n",
    "        if ubicacion:\n",
    "            # Definir la ruta del icono personalizado\n",
    "            icono_personalizado = folium.CustomIcon(icon_image=r'C:\\Users\\victo\\Pictures\\internet_icon.png', icon_size=(30, 30))\n",
    "            \n",
    "            # Agregar el marcador con el icono personalizado\n",
    "            folium.Marker(location=[ubicacion.latitude, ubicacion.longitude], popup=pais).add_to(marker_cluster)\n",
    "    except Exception as e:\n",
    "        print(f\"Error al geocodificar {pais}: {e}\")\n",
    "        continue\n",
    "# Guardar el mapa como un archivo HTML\n",
    "mapa.save('E:\\Cursos\\BC_Data_Science\\Repositorio\\ONLINE_DS_THEBRIDGE_V\\proyecto EDA\\mapa_geolocalizado_ips_destino_final_con_calor.html')\n",
    "\n",
    "# Mostrar el mapa en la notebook\n",
    "display(mapa)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
